scripts/object_hashing/create_object_md5.py

```diff
@@ -0,0 +1,121 @@
+    """Replace original version with tmp object which now has md5."""
+    
+    logging.info("Starting replace of the original object with tmp object to generate md5.")
+    copy_object(original_bucket_name, tmp_blob_name, original_bucket_name, original_blob_name, project_id)
+
+    # get the md5 and write to file
+    final_obj_bytes, final_obj_md5 = get_file_size(original_bucket_name, original_blob_name, project_id)
+    logging.info(f"Final object md5: {final_obj_md5}")
+
+    with open("object_md5.txt", "w") as md5_file:
+        md5_file.write(final_obj_md5)
+
+
+def create_tmp_object(original_bucket_name, original_blob_name, project_id=None):
+    """Create tmp copy of original object."""
+
+    tmp_blob_name = f"{original_blob_name}.tmp"
+    logging.info(f"Starting creation of tmp copy to: gs://{original_bucket_name}/{tmp_blob_name}")
+    cmd = f"gsutil cp -D gs://{original_bucket_name}/{original_blob_name} gs://{original_bucket_name}/{tmp_blob_name}"
+    subprocess.run(cmd, shell=True)
+
+    compare_file_sizes(original_bucket_name, original_blob_name, original_bucket_name, tmp_blob_name, project_id)
+
+    # replace original with tmp 
+    create_md5_object(original_bucket_name, original_blob_name, tmp_blob_name, project_id)
+
+
+def create_backup_object(backup_directory, original_bucket_name, original_blob_name, project_id=None):
+    """Make copy of object in provided location."""
+
+    # TODO: check for slash at end
+    logging.info("Backup directory has been provided.")
+    backup_bucket_name = backup_directory.split("/")[:-1][2] # fc-
+    backup_blob_name = "/".join(backup_directory.split("/")[3:]) + "".join(original_blob_name.split("/")[-1:])
+
+    logging.info(f"Starting creation of backup copy to: gs://{backup_bucket_name}/{backup_blob_name}")
+
+    copy_object(original_bucket_name, original_blob_name, backup_bucket_name, backup_blob_name, project_id)
+    compare_file_sizes(original_bucket_name, original_blob_name, backup_bucket_name, backup_blob_name, project_id)
+
+
+def compare_file_sizes(object_1_bucket, object_1_blob, object_2_bucket, object_2_blob, project_id=None):
+    """Check if 2 objects are the same size in bytes."""
+
+    obj_1_bytes, obj_1_md5 = get_file_size(object_1_bucket, object_1_blob, project_id)
+    obj_2_bytes, obj_2_md5 = get_file_size(object_2_bucket, object_2_blob, project_id)
+
+    logging.info(f"src  object size: {obj_1_bytes} bytes with md5 {obj_1_md5}")
+    logging.info(f"dest object size: {obj_2_bytes} bytes with md5 {obj_2_md5}")
+
+    if obj_1_bytes != obj_2_bytes:
+        raise ValueError(f"Copy of object failed - {object_1_blob} and {object_2_blob} do not have the same file size. \
+            This is likely a transient failure. Please submit workflow again.")
+    
+    logging.info(f"Copy of object complete - {object_1_blob} and {object_2_blob} have the same file size. \n")
+
+
+def get_file_size(bucket_name, object_name, project_id=None):
```

Comment:
I think you end up always calling this twice for the file, once when comparing objects to make sure they match and once when getting md5.